### ISSUE TITLE
Feature Request: Custom rake task name

### ISSUE THREAD

**Issue by PikachuEXE** [odlp/oops_a_rake#2]
I do want custom task name sometimes instead of having to adjust task class name...
```ruby
class GreetingTask
  include OopsARake::Task

  name "say_hello"
  description "An enthusiastic greeting"

  def call
    puts "Hello!"
  end
end
```

**Comment by odlp** [comment]
Hi @PikachuEXE,

I like this suggestion, it makes a lot of sense. 

Unfortunately adding this feature is quite complicated under the current implementation. At the moment the task is registered immediately when the `OopsARake::Task` module is included:

https://github.com/odlp/oops_a_rake/blob/b3be5025f7ea85e00069e3e4683537258b41e6b6/lib/oops_a_rake/task.rb#L5-L8

Originally I thought we could just rename the Rake task if `#name` is called, but Rake doesn't seem to have a public API for renaming tasks (it seems possible by [accessing instance variables](http://lslezak.blogspot.com/2009/06/renaming-rake-task.html), but I'd rather avoid that).

To support this feature I think we'd have to defer registration of Rake tasks until a hook is called, similar to this in the `Rakefile`:

```ruby
# Rakefile

require "oops_a_rake"

# Optional config
OopsARake.task_paths = ["lib/tasks/*.rb"]

OopsARake.load_tasks
```

I think this more explicit loading mechanism is probably a better design, similar to [`FactoryBot.find_definitions`](https://www.rubydoc.info/gems/factory_bot/FactoryBot.find_definitions).

I can take a look at adding this over the weekend. Ay feedback on the proposed changes would be welcome!

**Comment by PikachuEXE** [comment]
Or something like this (but not sure if can be done):
```ruby
class GreetingTask
  include OopsARake::Task::Lazy

  register do |task|
    #not calling `name` = generate name from class
    task.name "say_hello"
    task.description "An enthusiastic greeting"
    # other config
  end

  def call
    puts "Hello!"
  end
end

```

**Comment by odlp** [comment]
@PikachuEXE I've made a [PR](https://github.com/odlp/oops_a_rake/pull/3) proposing a solution like this:

```ruby
class ObscureClassNameTask do
  include OopsARake::Task.with_name("custom_name")
  
  def call
    puts "Hello"
  end
end
```

Would this work for you?

**Comment by PikachuEXE** [comment]
Might as well make it accept options like `dry-types`:
https://dry-rb.org/gems/dry-types/1.2/getting-started/
```ruby
  include OopsARake::Task(
    name: "custom_name",
  )
```
Their implementation looks like:
https://github.com/dry-rb/dry-types/blob/v1.4.0/lib/dry/types.rb#L201-L203

**Comment by odlp** [comment]
I don't think that'll work in this case unless I'm missing something. Mimicking the `self.Types` method from `dry-types` would mean that'd including the module would be `include OopsARake.Task` rather than `include OopsARake::Task`, which seems weird to me.

My preference would be to not compromise the base use case to support this feature. I'd suggest a halfway house which is more flexible than `::with_name` for future use cases:

```ruby
include OopsARake::Task.with_options(name: "custom_name")
```

**Comment by PikachuEXE** [comment]
As long as it's flexible enough I am fine with it

**Comment by odlp** [comment]
I've just released version `0.2.0` which supports custom task names. Thanks for the suggestion & feedback @PikachuEXE
